src/iuse_actor.cpp

```diff
@@ -808,26 +808,28 @@ void pick_lock_actor::load( JsonObject &obj )
     pick_quality = obj.get_int( "pick_quality" );
 }
 
-long pick_lock_actor::use( player &p, item &it, bool, const tripoint & ) const
+long pick_lock_actor::use( player &p, item &it, bool, const tripoint &pos) const
 {
     if( p.is_npc() ) {
         return 0;
     }
-    tripoint dirp;
-    if( !choose_adjacent( _( "Use your lockpick where?" ), dirp ) ) {
-        return 0;
-    }
-    if( dirp == p.pos() ) {
-        p.add_msg_if_player( m_info, _( "You pick your nose and your sinuses swing open." ) );
-        return 0;
-    }
+    tripoint dirp = pos;
     const ter_id type = g->m.ter( dirp );
-    if( g->critter_at<npc>( dirp ) ) {
-        p.add_msg_if_player( m_info,
-                             _( "You can pick your friends, and you can\npick your nose, but you can't pick\nyour friend's nose" ) );
-        return 0;
+    if( pos == p.pos() ) {
+        if( !choose_adjacent( _( "Use your lockpick where?" ), dirp ) ) {
+            return 0;
+        }
+        if( dirp == p.pos() ) {
+            p.add_msg_if_player( m_info, _( "You pick your nose and your sinuses swing open." ) );
+            return 0;
+        }
+        if( g->critter_at<npc>( dirp ) ) {
+            p.add_msg_if_player( m_info,
+                                _( "You can pick your friends, and you can\npick your nose, but you can't pick\nyour friend's nose" ) );
```

Comment:
Not related to this PR, but why are these linebreaks here in the first place? Might want to remove them since the code does automatical string folding when printing the messages.

Reply:
Probably some kind of a joke.